src/OpenApi/src/Services/OpenApiComponentService.cs

```diff
@@ -42,7 +42,26 @@ private readonly ConcurrentDictionary<(Type, ParameterInfo?), JsonObject> _schem
     {
         OnSchemaGenerated = (context, schema) =>
         {
-            schema.ApplyPrimitiveTypesAndFormats(context.TypeInfo.Type);
+            var type = context.TypeInfo.Type;
+            // Fix up schemas generated for IFormFile, IFormFileCollection, Stream, and PipeReader
+            // that appear as properties within complex types.
+            if (type == typeof(IFormFile) || type == typeof(Stream) || type == typeof(PipeReader))
```

Comment:
What happens if a complex type has a property of type `MemoryStream`?  Does that get the same magic handling during model binding?

Reply:
Nope, we don't have any special handling for `MemoryStream`. Both MVC/minimal will try to bind the MemoryStream property as a complex object (e.g. read me from JSON). This doesn't make a ton of sense but I'd argue that if someone is trying to consume a `MemoryStream` as input to an HTTP API then there's no sensible representation for it.

Reply:
To be more explicit, we're specifically not handling subtypes of those magic types?

Reply:
> To be more explicit, we're specifically not handling subtypes of those magic types?

Correct -- we only handle `Stream` and `PipeReader` because there are model binding implementations for those.